drivers/flow.c

```diff
@@ -0,0 +1,221 @@
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" 
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE 
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF 
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+ * POSSIBILITY OF SUCH DAMAGE.
+ ******************************************************************************/
+
+/*******************************************************************************
+ * \file flow.c
+ * 
+ * \author MAV'RIC Team
+ * \author Julien Lecoeur
+ *
+ * \brief Driver for optic flow sensors
+ *
+ ******************************************************************************/
+
+#include "flow.h"
+#include "time_keeper.h"
+
+
+//------------------------------------------------------------------------------
+// PRIVATE FUNCTIONS IMPLEMENTATION
+//------------------------------------------------------------------------------
+
+/**
+ * \brief Swap bytes iof signed 16 bits integer
+ */
+static inline int16_t endian_rev16(int16_t data)
+{
+	return ((data >> 8) & 0x00ff) | ((data & 0x00ff) << 8);
+};
+
+
+//------------------------------------------------------------------------------
+// PUBLIC FUNCTIONS IMPLEMENTATION
+//------------------------------------------------------------------------------
+
+bool flow_init(flow_t* flow, int32_t UID, usart_config_t usart_conf)
+{
+	bool success = true;
+
+	// Init members
+	flow->last_update_us  = time_keeper_get_micros();
+	flow->handshake_state = FLOW_NO_HANDSHAKE; 
+	flow->of_count 	      = 0;
+
+	// Init uart and buffers
+	uart_int_set_usart_conf(UID, &usart_conf);
+	uart_int_init(UID);
+	buffer_make_buffered_stream(&(flow->uart_buffer_in), &(flow->uart_stream_in));
+	buffer_make_buffered_stream(&(flow->uart_buffer_out), &(flow->uart_stream_out));
+	uart_int_register_read_stream(uart_int_get_uart_handle(UID), &(flow->uart_stream_in));
+	uart_int_register_write_stream(uart_int_get_uart_handle(UID), &(flow->uart_stream_out));
+	
+	// Init MAVLink stream
+	mavlink_stream_conf_t mavlink_stream_conf =
+	{
+		.sysid       = 1,
+		.compid      = 50,
+		.debug       = true,
+	};
+	success &= mavlink_stream_init(	&(flow->mavlink_stream),
+					&mavlink_stream_conf,
+					&(flow->uart_stream_in),
+					&(flow->uart_stream_out));
+
```

Comment:
Issue opened.

However you added the debug message only in the mavlink_stream init, and not in the flow_init